File: src/main/java/com/hwscs/backend/service/interfaces/NurseService.java
package com.hwscs.backend.service.interfaces;

import com.hwscs.backend.dto.response.NurseResponseDto;
import com.hwscs.backend.dto.response.NurseShiftResponseDto;

import java.util.List;

public interface NurseService {

    // Nurse views their own profile
    NurseResponseDto getMyProfile(String username);

    // Nurse views their own upcoming/all shifts
    List<NurseShiftResponseDto> getMyShifts(String username);

    // Incharge/DutyOfficer: list all nurses in a department
    List<NurseResponseDto> getNursesByDepartment(Integer departmentId);

    // View a single nurse by ID (Incharge/DutyOfficer)
    NurseResponseDto getNurseById(Integer nurseId);
}

File: frontend/src/app/core/services/nurse.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { NurseResponseDto, NurseShiftResponseDto } from '../models/models';

@Injectable({
  providedIn: 'root'
})
export class NurseService {
  private readonly apiUrl = '/api/nurses';

  constructor(private http: HttpClient) {}

  getMyProfile(): Observable<NurseResponseDto> {
    return this.http.get<NurseResponseDto>(`${this.apiUrl}/me`);
  }

  getMyShifts(): Observable<NurseShiftResponseDto[]> {
    return this.http.get<NurseShiftResponseDto[]>(`${this.apiUrl}/me/shifts`);
  }

  getNursesByDepartment(departmentId: number): Observable<NurseResponseDto[]> {
    return this.http.get<NurseResponseDto[]>(`${this.apiUrl}/department/${departmentId}`);
  }
}
